packages/blade/src/components/Input/SelectInput/SelectInput.tsx

```diff
@@ -35,6 +35,16 @@ type SelectInputProps = Pick<
   | 'testID'
 > & {
   icon?: IconComponent;
+  /**
+   * Controlled value of the Select. Use it in combination of `onChange`.
+   *
+   * Check out [Controlled Dropdown Documentation](https://blade.razorpay.com/?path=/story/components-dropdown-with-select--controlled-dropdown&globals=measureEnabled:false) for example.
+   */
+  value?: string | string[];
```

Comment:
should it be `value` or `values`? same thing for `defaultValue`

Reply:
Good question. This is why you're principle engineer 🤭🫡

I think it depends on what we do here - https://github.com/razorpay/blade/pull/1139#discussion_r1188142533

I am too feeling `values` might be better option now. The reason we had `value` was because this code seems a bit complex for single select usecases-
```jsx
const [currentSelection, setCurrentSelection] = React.useState()

return (
  <Dropdown>
    <SelectInput 
      values={[currentSelection]} 
      onChange={({values}) => setCurrentSelection(values[0])}
    />
    <DropdownOverlay>
      <ActionList />
    </DropdownOverlay>
  </Dropdown>
)
```

You can also store array in useState and directly set values and onChange values but then on consumer end when they are programatically selecting something, they will have to ensure that array only has one value.



But... since we already decided to have `values` in `onChange` for the similar reason, I feel we'll have to be consistent with that decision.

Reply:
Turning into `values` to be consistent

Reply:
Leaning towards using `value`. I think `value` should be fine whether it's a `string` or `string[]`. This would be analogous with `event.target.value` and also we're using `value` prop in input fields 🤔 

A slight inconvenience, otherwise for some components it would be `value` and for some `values`.

Reply:
That is true. Also, A single select with plain values is a lot more common usecase than multiselect. So if we create API that is more multiselect-friendly, it would cause inconvenience to a lot more people.